File: com/pdfjet/SVGPath.java
/*
 * SVGPath.java
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.pdfjet;

import java.util.List;

/**
 * A path or a shape of an SVG file, as PDF path operations in the space of
 * the svg element, and what it is drawn with.
 */
class SVGPath {
    /** The default constructor */
    SVGPath() {
    }

    List<PathOp> operations;        // The PDF path operations
    int fill = Color.transparent;   // The fill color, or transparent for none
    int stroke = Color.transparent; // The stroke color, or transparent for none
    float strokeWidth = 0f;         // The stroke width, in the space of the svg element
    boolean evenOdd = false;        // Filled with the even-odd rule
    CapStyle lineCap = CapStyle.BUTT;
    JoinStyle lineJoin = JoinStyle.MITER;
    float fillAlpha = 1f;
    float strokeAlpha = 1f;
}

File: Sources/PDFjet/SVGPath.swift
/**
 * SVGPath.swift
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
import Foundation

/// A path or a shape of an SVG file, as PDF path operations in the space of
/// the svg element, and what it is drawn with.
class SVGPath {
    var operations = [PathOp]()             // The PDF path operations
    var fill: Int32 = Color.transparent     // The fill color, or Color.transparent for none
    var stroke: Int32 = Color.transparent   // The stroke color, or Color.transparent for none
    var strokeWidth: Float = 0.0            // The stroke width, in the space of the svg element
    var evenOdd = false                     // Filled with the even-odd rule
    var lineCap = CapStyle.BUTT
    var lineJoin = JoinStyle.MITER
    var fillAlpha: Float = 1.0
    var strokeAlpha: Float = 1.0
}
